### PR TITLE
feat: new Txs Tables

### PR DESCRIPTION
**Pull request by songwongtp**
## Describe your changes
- Add relation column
- Add relation filter
- Refactor txs table

### CODE REVIEW

File: src/lib/services/txService.ts [hunk 1]
```diff
@@ -8,22 +8,20 @@ import { useCallback } from "react";
...
-  getExecuteTxsByContractAddressPagination,
-  getExecuteTxsCountByContractAddress,
-  getTxsByContractAddressPagination,
-  getTxsCountByContractAddress,
-} from "lib/query";
-import { MsgFurtherAction } from "lib/types";
-import type { Addr, ContractAddr, Option, Transaction } from "lib/types";
+  GetTxsByAddressPaginationDocument,
+  GetTxsCountByAddressDocument,
+} from "lib/gql/graphql";
```
Comment (evilpeach) [inline src/lib/services/txService.ts:13]: We shouldn't import from `lib/gql/graphql`

File: src/lib/services/txService.ts [hunk 2]
```diff
@@ -56,188 +54,117 @@ export const useTxData = (txHash: Option<string>): UseQueryResult<TxData> => {
   });
 };
 
-export const useExecuteTxsByContractAddressPagination = (
-  contractAddress: ContractAddr,
+export const useTxsByAddressPagination = (
+  address: Option<Addr>,
+  search: string,
+  filters: TxFilters,
+  relation: Option<boolean>,
```
Comment (evilpeach) [inline src/lib/services/txService.ts:61]: isRelated?

File: src/lib/services/txService.ts [hunk 3]
```diff
@@ -56,188 +54,117 @@ export const useTxData = (txHash: Option<string>): UseQueryResult<TxData> => {
...
+      .request(GetTxsByAddressPaginationDocument, {
+        expression,
         offset,
         pageSize,
       })
-      .then(({ contract_transactions_view }) =>
+      .then(({ account_transactions }) =>
         /**
          * @remarks because contract_transactions_view is view table, all fields can be undefined by type
          */
```
Comment (evilpeach) [inline src/lib/services/txService.ts:78]: remove

File: src/lib/services/txService.ts [hunk 4]
```diff
@@ -56,188 +54,117 @@ export const useTxData = (txHash: Option<string>): UseQueryResult<TxData> => {
...
+        account_transactions.map((tx) => ({
+          hash: parseTxHash(tx.transaction.hash),
+          messages: snakeToCamel(tx.transaction.messages),
+          sender: tx.transaction.account.address as Addr,
+          isSigner: tx.is_signer,
+          height: tx.block.height,
+          created: parseDateOpt(tx.block.timestamp),
+          success: tx.transaction.success,
           actionMsgType: getActionMsgType([
             /* these value must not be null */
```
Comment (evilpeach) [inline src/lib/services/txService.ts:88]: remove

File: src/lib/services/txService.ts [hunk 5]
```diff
@@ -56,188 +54,117 @@ export const useTxData = (txHash: Option<string>): UseQueryResult<TxData> => {
...
          * @remarks because contract_transactions_view is view table, all fields can be undefined by type
          */
-        contract_transactions_view.map((transaction) => ({
-          hash: parseTxHash(transaction.hash),
-          messages: snakeToCamel(transaction.messages),
-          sender: transaction.sender as Addr,
-          height: transaction.height,
-          created: parseDateOpt(transaction.timestamp),
-          success: transaction.success,
+        account_transactions.map((tx) => ({
```
Comment (evilpeach) [inline src/lib/services/txService.ts:79]: Turn `tx` to `{transaction}`.

File: src/lib/pages/past-txs/types.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import type { Option, TxFilters } from "lib/types";
+
+export interface PastTxsState {
+  search: string;
+  filters: TxFilters;
+  relation: Option<boolean>;
```
Comment (evilpeach) [inline src/lib/pages/past-txs/types.ts:6]: isRelated or isSigner?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/components/table/transactions/TransactionsTableRow.tsx [hunk 7]
```diff
@@ -82,6 +91,12 @@ export const TransactionsTableRow = ({
           </Flex>
         </TableRow>
 
+        {showAction && (
```
Comment (bkioshn) [inline src/lib/components/table/transactions/TransactionsTableRow.tsx:94]: in 
```
<AccordionTx
    key={index.toString() + msg.type}
    message={msg}
    allowFurtherAction={false}
/>
))}
```
change `allowFurtherAction={false}` to `allowFurtherAction={showAction}`

**Review by bkioshn — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by bkioshn — APPROVED** [review]

**Pull request merged**